src/sshfs_mount/sshfs_mount_handler.cpp

```diff
@@ -60,229 +48,162 @@ void start_and_block_until(mp::Process* process, Signal signal, std::function<bo
+            }
 
-            install_sshfs_for(vm->vm_name, session, on_install, timeout);
+            install_sshfs_for(vm->vm_name, session, timeout);
         },
         server);
 
-    auto config = sshfs_server_configs[vm->vm_name][target_path];
     // Can't obtain hostname/IP address until instance is running
     config.host = vm->ssh_hostname();
+    process = platform::make_sshfs_server_process(config);
 
-    auto sshfs_server_process_t = mp::platform::make_sshfs_server_process(config);
-    // FIXME: ProcessFactory really should return qt_delete_later_unique_ptr<Process> as Process emits signals
-    // and the respective slots may be called on the event loop, but unique_ptr can delete the Process before
-    // the slots are fired, causing a crash.
-    mp::qt_delete_later_unique_ptr<mp::Process> sshfs_server_process(sshfs_server_process_t.release());
-
-    QObject::connect(
-        sshfs_server_process.get(), &mp::Process::finished, this,
-        [this, instance = vm->vm_name, target_path](mp::ProcessState exit_state) {
-            if (exit_state.completed_successfully())
-            {
-                mpl::log(mpl::Level::info, category,
-                         fmt::format("Mount '{}' in instance \"{}\" has stopped", target_path, instance));
-            }
-            else
-            {
-                mpl::log(mpl::Level::warning, // not error as it failing can indicate we need to install sshfs in the VM
-                         category,
-                         fmt::format("Mount '{}' in instance \"{}\" has stopped unexpectedly: {}", target_path,
-                                     instance, exit_state.failure_message()));
-            }
-
-            mount_processes[instance].erase(target_path);
-        });
-
-    QObject::connect(
-        sshfs_server_process.get(), &mp::Process::error_occurred, this,
-        [instance = vm->vm_name, target_path, process = sshfs_server_process.get()](QProcess::ProcessError error,
-                                                                                    QString error_string) {
-            mpl::log(mpl::Level::error, category,
-                     fmt::format("There was an error with sshfs_server for instance \"{}\" with path '{}': {} - {}",
-                                 instance, target_path, mp::utils::qenum_to_string(error), error_string));
-        });
+    QObject::connect(process.get(), &Process::finished, [this](const ProcessState& exit_state) {
+        if (exit_state.completed_successfully())
+        {
+            mpl::log(mpl::Level::info, category,
+                     fmt::format("Mount \"{}\" in instance '{}' has stopped", target, vm->vm_name));
+        }
+        else
+        {
+            // not error as it failing can indicate we need to install sshfs in the VM
+            mpl::log(mpl::Level::warning, category,
+                     fmt::format("Mount \"{}\" in instance '{}' has stopped unexpectedly: {}", target, vm->vm_name,
```

Comment:
I find this message misleading. The process completing "unsuccessfully" and "unexpectedly" are two different things. It may complete unsuccessfully after it is killed, or even terminated (e.g. if it encountered some error while quitting); and, as far as we know, it could stop unexpectedly without signalling any error.